### PR TITLE
Introduce event filter in CloudEventSource

### PR DESCRIPTION
**Pull request by SpiritZhou**
Introduce event filter in CloudEventSource

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #
https://github.com/kedacore/keda/pull/5424

### CODE REVIEW

File: content/docs/2.14/operate/events.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
-+++
+[+++
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:1]: This is probably not intentional?
```suggestion
+++
```

File: content/docs/2.14/operate/events.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 title = "Events"
 description = "Kubernetes Events emitted by KEDA"
 weight = 100
 +++
 
 ## Kubernetes Events emitted by KEDA
 
-KEDA emits the following [Kubernetes Events](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#event-v1-core):
+KEDA emits the following \[Kubernetes Events\](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#event-v1-core):
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:9]:
```suggestion
KEDA emits the following [Kubernetes Events](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#event-v1-core):
```

File: content/docs/2.14/operate/events.md [hunk 3]
```diff
@@ -74,7 +74,7 @@ There will be multiple types of destination to emit KEDA events to.
 
 Here is an overview of the supported destinations:
 
-- [HTTP endpoint](#http-endpoint).
+- \[HTTP endpoint\](#http-endpoint).
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:77]:
```suggestion
- [HTTP endpoint](#http-endpoint).
```

File: content/docs/2.14/operate/events.md [hunk 4]
```diff
@@ -86,4 +86,4 @@ Here is an overview of the supported destinations:
 | Event Type                                | Scenario Description                                                                                                                 | 
 | ------------------------------------- |  --------------------------------------------------------------------------------------------------------------------------- | 
 | `keda.scaledobject.ready.v1`                   | On the first time a ScaledObject is ready, or if the previous ready condition status of the object was `Unknown` or `False` |  
-| `keda.scaledobject.failed.v1`             | If the check validation for a ScaledObject fails |      
\ No newline at end of file
+| `keda.scaledobject.failed.v1`             | If the check validation for a ScaledObject fails |      ](../../2.13/operate/events.md)
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:89]: Is this intentional?
```suggestion
| `keda.scaledobject.failed.v1`             | If the check validation for a ScaledObject fails |     
```

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
I'm not sure I get this PR, are we missing changes for filtering, @SpiritZhou ?

**Comment by SpiritZhou** [comment]
> I'm not sure I get this PR, are we missing changes for filtering, @SpiritZhou ?

Sorry, I might have messed up when merging main branch.

File: content/docs/2.14/operate/events.md [hunk 5]
```diff
@@ -49,6 +49,11 @@ spec:
   destination:
     http:
       uri: http://foo.bar
+  eventSubscription: #Optional. Submit included/excluded event types will filter events when emitting events. 
+        includedEventTypes: #Optional. Only events in this section will be emitted.
+        - keda.scaledobject.failed.v1
+        excludedEventTypes: #Optional. Events in this section will not be emitted.       
+        - keda.scaledobject.ready.v1
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:56]: Nit - I think the formatting might be off?

```suggestion
  destination:
    http:
      uri: http://foo.bar
  eventSubscription: #Optional. Submit included/excluded event types will filter events when emitting events. 
    includedEventTypes: #Optional. Only events in this section will be emitted.
    - keda.scaledobject.failed.v1
    excludedEventTypes: #Optional. Events in this section will not be emitted.       
    - keda.scaledobject.ready.v1
```

**Review by tomkerkhove — APPROVED** [review]
LGTM, thanks! Waiting for feature to merge

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**